src/main/java/org/wso2/siddhi/plugins/idea/debugger/SiddhiDebugProcess.java

```diff
@@ -0,0 +1,687 @@
+
+            int offset = breakpointPosition.getOffset();
+            PsiElement element = null;
+            if (psiFile != null) {
+                element = psiFile.findElementAt(offset);
+            }
+            if ((element == null)) {
+                return;
+            }
+            PsiElement nextVisibleSibling = getNextVisibleSiblingSkippingComments(element);
+            if (!(isAQueryOutputBeginningKeyword(((LeafPsiElement) element).getElementType())
+                    //TODO:once the antlr tree collapsing issue fixed remove one getParent() from below
+                    && element.getParent().getParent().getParent().getParent() instanceof ExecutionElementNode
+                    && PsiTreeUtil.getParentOfType(element, QueryOutputNode.class) != null
+                    && PsiTreeUtil.getParentOfType(nextVisibleSibling, QueryOutputNode.class) != null
+                    && PsiTreeUtil.getParentOfType(element, PartitionNode.class) == null)) {
+                return;
+            }
+            outBreakpoints.add(breakpoint);
+            sendBreakpoints();
+            getSession().updateBreakpointPresentation(breakpoint, AllIcons.Debugger.Db_verified_breakpoint, null);
+        }
+
+        private boolean isAQueryOutputBeginningKeyword(IElementType elementType) {
+            return elementType == SiddhiTypes.INSERT || elementType == SiddhiTypes.UPDATE
+                    || elementType == SiddhiTypes.DELETE || elementType == SiddhiTypes.RETURN;
+        }
+
+        @Override
+        public void unregisterBreakpoint(@Nonnull XLineBreakpoint<SiddhiBreakpointProperties> breakpoint,
+                                         boolean temporary) {
+            XSourcePosition breakpointPosition = breakpoint.getSourcePosition();
+            if (breakpointPosition == null) {
+                return;
+            }
+            outBreakpoints.remove(breakpoint);
+            sendRemovedBreakpoint(breakpoint);
+        }
+
+        void sendRemovedBreakpoint(@Nonnull XLineBreakpoint<SiddhiBreakpointProperties> breakpoint) {
+            XSourcePosition breakpointPosition = breakpoint.getSourcePosition();
+            if (breakpointPosition == null) {
+                return;
+            }
+            VirtualFile file = breakpointPosition.getFile();
+            if (!file.getPath().replaceAll("/", Matcher.quoteReplacement(File.separator)).equalsIgnoreCase
+                    (myDebugFilePath)) {
+                return;
+            }
+            int line = breakpointPosition.getLine();
+            String name = file.getName();
+            String terminal = "OUT";
+            int queryIndex = -1;
+            for (int j = 0; j < queryOutLinePositions.size(); j++) {
+                if ((queryOutLinePositions.get(j) - 1) == line) {
+                    queryIndex = j;
+                    break;
+                }
+            }
+            String stringBuilder = "{\"command\":\"" + Command.REMOVE_BREAKPOINT +
```

Comment:
Change this as well according to the above comment.

Reply:
Fixed in commit 3aa87a6